### DISCUSSION TITLE
JSON getting queried incorrectly, maybe a bug

### DISCUSSION THREAD

**Discussion by bgigurtsis** [grafana/grafana-infinity-datasource#494] [Q&A] [answered]
Hello and happy holidays,

I have two Grafana panels querying two different JSON files using an Infinity data source. The two panels JSON is exactly the same, and the JSON files themselves are quite similar, only with some added data. The thing that I am querying (start_time) is in the same place in each file and so the same JSON query should work. 

The location of the JSON I want to query, and thus the query scan[start_time], doesn't change between the two JSON files. The same JSON query should work for both files. However with the second file the query does not work. It seems like infinity thinks the root 'scan' is inside 'primary_identifiers' as when I query scan[type] it returns 'semgrep_id'. Querying scan[start_time], and formatted as timestamp, returns invalid date. I've tried using a third party website and the same query definitely works for both JSON files.

I've tried looking at the JSON that's returned by Infinity when it queries my data source. The data returned for the JSON file where scan[start_time] works looks normal and is not too long, but for the file where scan[start_time] doesn't work the data returned is about 7000 lines long and it seems like that might be the problem but im not sure.

This is the JSON (sanitised) where it seems to be broken, and scan[start_time] doesn't work:

```
{
  "version": "14.0.4",
  "vulnerabilities": [
    {
      "id": "123",
      "category": "sast",
      "message": "Test",
      "description": "test",
      "cve": "test",
      "severity": "Medium",
      "scanner": {
        "id": "semgrep",
        "name": "Semgrep"
      },
      "location": {
        "file": "src.ts",
        "start_line": 12
      },
      "identifiers": [
        {
          "type": "semgrep_id",
          "name": "test",
          "value": "test",
          "url": "test"
        },
        {
          "type": "test",
          "name": "test",
          "value": "test"
        }
      ]
    }
  ],
  "dependency_files": [],
  "scan": {
    "analyzer": {
      "id": "semgrep",
      "name": "Semgrep",
      "url": "https://gitlab.com/gitlab-org/security-products/analyzers/semgrep",
      "vendor": {
        "name": "GitLab"
      },
      "version": "3.10.0"
    },
    "scanner": {
      "id": "semgrep",
      "name": "Semgrep",
      "url": "grep",
      "vendor": {
        "name": "GitLab"
      },
      "version": "0.121.2"
    },
    "primary_identifiers": [
      {
        "type": "test",
        "name": "test",
        "value": "000"
      },
      {
        "type": "semgrep_id",
        "name": "test",
        "value": "111"
      }
    ],
    "type": "sast",
    "start_time": "2022-12-14T10:47:25",
    "end_time": "2022-12-14T10:47:43",
    "status": "success"
  }
}

```

This is the JSON where querying scan[start_time] actually works and returns 2022-12-14 16:14:37":

```
{
   "version":"14.0.4",
   "vulnerabilities":[
      
   ],
   "scan":{
      "analyzer":{
         "id":"secrets",
         "name":"secrets",
         "url":"https://gitlab.com/gitlab-org/security-products/analyzers/secrets",
         "vendor":{
            "name":"GitLab"
         },
         "version":"4.5.4"
      },
      "scanner":{
         "id":"gitleaks",
         "name":"Gitleaks",
         "url":"https://github.com/zricethezav/gitleaks",
         "vendor":{
            "name":"GitLab"
         },
         "version":"8.15.0"
      },
      "type":"secret_detection",
      "start_time":"2022-12-14T16:14:37",
      "end_time":"2022-12-14T16:14:37",
      "status":"success"
   }
}
```
I've also attached screenshots of my panel setup and the various queries. Let me know if you need any more info. 

<img width="1106" alt="Screenshot 2022-12-22 at 14 44 33" src="https://user-images.githubusercontent.com/69985092/209161442-358d102e-09cb-4036-90d6-24e5a7cc5db7.png">
<img width="1082" alt="Screenshot 2022-12-22 at 14 44 22" src="https://user-images.githubusercontent.com/69985092/209161455-ba08f1e1-af8b-4ead-93dc-e31b94ee4e6f.png">

Any ideas? Thank you!

**Comment by yesoreyeram** [comment, marked as answer]
Probably a bug in the version you are using. Suggesting to update the plugin version and also suggest using "backend" parser option which is available in latest version of plugin.

**Reply by bgigurtsis** [reply]
> Probably a bug in the version you are using. Suggesting to update the plugin version and also suggest using "backend" parser option which is available in latest version of plugin.

Thank you for the quick reply, that is a good bet. The Infinity plugin is on version 0.8.8, and my Grafana is on v8.3.0. Do you know if there's a minimum Grafana version for the newest Infinity plugin? I installed the plugin yesterday and when I go to the plugin on Grafana I'm not seeing an update button (I am an admin on Grafana).

**Reply by yesoreyeram** [reply]
AFAIK, 8.4.7 is the required version specified in the plugin. But that's not a hard requirement.

**Reply by bgigurtsis** [reply]
Understood, thank you again
